lib/segment/src/vector_storage/sparse_raw_scorer.rs

```diff
@@ -98,28 +109,16 @@ impl<'a, TVectorStorage: SparseVectorStorage> RawScorer for SparseRawScorer<'a,
         let scores = points
             .take_while(|_| !self.is_stopped.load(Ordering::Relaxed))
             .filter(|point_id| self.check_vector(*point_id))
-            .map(|point_id| {
-                let vector = self.vector_storage.get_sparse(point_id);
-                ScoredPointOffset {
-                    idx: point_id,
-                    score: vector.score(&self.query),
-                }
-            });
+            .filter_map(|point_id| self.score_overlapping_point(point_id));
```

Comment:
I guess we have a bug here. If the search query vector does not overlap everything in dataset, the user will get an empty result instead of some `TOP` zero-scored vectors.
I didn't get an idea why do we need such filtering and overlap checks

Reply:
We need such filter because the inverted index does not return points which have zero overlaps in dimension with the query.

The goal is to make index and plain search equivalent.